src/sentry/integrations/vercel/uihook.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import
+
+from django.http import HttpResponse, HttpResponseNotFound
+from django.views.decorators.csrf import csrf_exempt
+
+from sentry.api.base import Endpoint
+from sentry.models import Integration, Organization
+from sentry.web.api import allow_cors_options
+from sentry.web.helpers import render_to_response
+
+import json
+
+
+class VercelUIHook(Endpoint):
+    auth_required = False
+
+    @csrf_exempt
+    @allow_cors_options
+    def dispatch(self, request):
+        if request.method == "OPTIONS":
+            return HttpResponse(status=200)
+        if request.method != "POST":
+            return HttpResponseNotFound("Page not found")
+        body_unicode = request.body.decode("utf-8")
+        body = json.loads(body_unicode)
+        configuration_id = body["configurationId"]
+        user_id = body["user"]["id"]
+        team_id = body["teamId"]
+        integration = Integration.objects.get(external_id=user_id or team_id, provider="vercel")
+        if integration and integration.status == 0:
+            organization = Organization.objects.get(
+                id=integration.metadata["configurations"][configuration_id]["organization_id"]
```

Comment:
@ceorourke We should make this robust enough to not result in a `KeyError` if for some reason the `configuration_id` key doesn't exist on `integration.metadata["configurations"]`. In that case, we should do the same `The requested integration does not exist.` error as below. Also, you should check the `OrganizationIntegration` table to see if a row with the`organization_id` and `integration_id` that matches the data here exists. That means that's actually installed on that organization* Only if that exists should we render a response with success. 

*Yes, we are using this hacky `metadata.configurations` to find the org here based off the `configuration_id` but we should still check the installation the "normal" way with the `OrganizationIntegration` table.